src/GraphQL.Client/UriExtensions.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+
+namespace GraphQL.Client.Http
+{
+    public static class UriExtensions
+    {
+        /// <summary>
+        /// Returns true if <see cref="Uri.Scheme"/> equals "wss" or "ws"
+        /// </summary>
+        /// <param name="uri"></param>
+        /// <returns></returns>
+        public static bool HasWebSocketScheme(this Uri uri) => uri.Scheme.Equals("wss") || uri.Scheme.Equals("ws");
+
+        /// <summary>
+        /// Infers the websocket uri from <paramref name="uri"/>.
+        /// </summary>
+        /// <param name="uri"></param>
+        /// <returns></returns>
+        public static Uri GetWebSocketUri(this Uri uri)
+        {
+            if (uri.HasWebSocketScheme())
+                return uri;
+
+            string webSocketScheme = uri.Scheme == "https" ? "wss" : "ws";
```

Comment:
Comparison needs to be case-insensitive. Also, why default to `"ws"` when the current scheme is not `"https"`? For example, `"ftp"` is a valid URI scheme, but it should not be converted to `"ws"`. Instead, I would expect this method to throw an exception when it is not a HTTP(S) scheme.

Reply:
done it properly in #279